### DISCUSSION TITLE
STM32H743VI Built-in Networking Compatibility

### DISCUSSION THREAD

**Discussion by chrisshift** [cesanta/mongoose#2863] [Ask questions, get answers] [answered]
I am working on a potential commercial application and have run into a hurdle getting the STM32H743VI chip to work with the built in networking. The code was originally developed on a H743ZI Nucleo board and has since been ported to the H743VI used on a custom PCB. The PHY used is a LAN8742a as on the Nucleo and the Magjack used is a Hanrun HY911105AE.

The code is developed in CUBEIDE from the nucleo H743ZI examples.

The LEDs on the jack indicate a connection when the cable is connected. Yellow LED, which flashes occasionally indicating activity and the Green LED indicating A 100mb connection. This is as per the datasheet: 

_"When configured in the default Link/Activity mode, the LED1 output is driven active whenever the device detects a valid
link, and blinks when CRS is active (high) indicating activity.
When configured in the default Link Speed mode, the LED2 output is driven active when the operating speed is 100
Mbps. This LED will go inactive when the operating speed is 10 Mbps or during line isolation."_

Mongoose confirms that it has a 100mb connection and when the cable is connected it starts receiving packets, however the board does not respond to a ping and any packets sent from mongoose do not show up on Wireshark.

The board has also been configured with a minimal code setup to no avail as per the following tutorial: https://mongoose.ws/documentation/tutorials/stm32/all-cube-baremetal-builtin/

Can it be confirmed that the H743VI should work as expected with the built-in driver and that the problem is most likely hardware based? Could anyone shed some light as to where I should be looking or digging to solve this?

**Comment by scaprile** [comment, marked as answer]
Our tutorials are for the boards that we tested, those shown in the tutorial.
If you design your own hardware, you have to setup your chips according to your hardware.
There is an example for the H743, hence our driver works for the H743
The difference between ZI and VI is the pin count (RM section 9)
RJ-45 LEDs are operated on by the PHY
You don't mention whether you see your link is full or half-duplex.
You don't mention what your hardware is, which pins you use
You don't mention whether you are clocking the PHY or being clocked by it.
My best guess is that you are missing the Tx clock, hence you are unable to transmit even though the MAC controller is working fine.

Since you mention you are a commercial prospect, I suggest you contact Cesanta and ask for guidance; will be glad to help you get your project on track. https://mongoose.ws/contact/

**Reply by chrisshift** [reply]
Thanks for the detailed response Scaprile. I have managed to solve the issue, although it's left me slightly confused. 
Setting the maximum output speed for all of the ethernet pins to Very High got everything working. What confuses me is that in the nucleo examples and all of my previous code, all the pins are left at their default of Low while working just fine.

To answer your previous questions, just for the sake of clarification to anyone reading this and facing any issues:
Link is reported as Full-duplex
PHY Pinout is as follows:
![Eth pinout](https://github.com/user-attachments/assets/5e24e243-f6d9-46e7-895d-035744e7af28)

PHY is provided with a 25MHz crystal. PHY produces 50MHz clock which is provided to MAC on pin PA1.

I'll be sure to contact Cesanta for any more guidance, thank you.

**Reply by scaprile** [reply]
Well, Ethernet certainly does require removing slew-rate limiters, we do that in all our baremetal examples. Regarding _our_ Cube tutorials, they recommend to start from a board setting, in order to have all pins setup according to their function with less effort, so pins should follow Cube/MX defaults.
